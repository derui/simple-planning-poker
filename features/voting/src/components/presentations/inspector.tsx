import { Icon, Icons } from "@spp/ui-icon";
import { clsx } from "clsx";

export interface Props {
  name: string;
}

const styles = {
  root: clsx("flex", "flex-col", "p-3", "items-center"),
  name: clsx("flex-auto", "text-indigo-700", "font-bold", "mb-2"),
} as const;

export const Inspector = function Inspector({ name }: Props) {
  return (
    <span className={styles.root}>
      <span className={styles.name}>{name}</span>
      <Icon type={Icons.user} size="xl" variant="indigo" />
    </span>
  );
};
