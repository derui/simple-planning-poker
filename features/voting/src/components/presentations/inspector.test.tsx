import { test, afterEach, expect } from "vitest";
import { render, cleanup } from "@testing-library/react";
import { Inspector } from "./inspector.js";

afterEach(cleanup);

test("should be able to render", async () => {
  const ret = render(<Inspector name="foo" />);

  expect(ret.container).toMatchSnapshot();
});
