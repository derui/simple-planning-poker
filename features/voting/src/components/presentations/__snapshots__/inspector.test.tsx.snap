// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be able to render 1`] = `
<div>
  <span
    class="flex flex-col p-3 items-center"
  >
    <span
      class="flex-auto text-indigo-700 font-bold mb-2"
    >
      foo
    </span>
    <span
      class="flex before:bg-indigo-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/user.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-10 before:h-10 w-10 h-10"
    />
  </span>
</div>
`;
